indico/modules/events/registration/client/js/form/fields/SingleChoiceInput.jsx

```diff
@@ -68,73 +79,85 @@ function SingleChoiceDropdown({
             (placesUsed[selectedChoice.id] || 0) - (existingValue[selectedChoice.id] || 0) >=
               selectedChoice.placesLimit)
         }
-        value={value[selectedChoice.id]}
-        onChange={(_evt, data) => onChange({[selectedChoice.id]: data.value})}
-        fluid
+        value={String(selectedSeats)}
+        onChange={evt => {
+          const selectedSlots = Number(evt.target.value);
+          onChange({[selectedChoice.id]: selectedSlots});
+        }}
+        aria-label={Translate.string('Select extra slots')}
+        required
+        {...comboBoxExtraProps}
       />
     );
   }
 
-  const options = choices.map(c => ({
-    key: c.id,
-    value: c.id,
-    disabled:
-      !c.isEnabled ||
-      isPaidChoiceLocked(c) ||
-      (c.placesLimit > 0 && (placesUsed[c.id] || 0) >= c.placesLimit && !existingValue[c.id]),
-    text: c.caption,
-    content: (
-      <div styleName="dropdown-text">
+  const options = choices.map(choice => ({
+    value: choice.caption,
+    label: (
+      <div styleName="dropdown-text" key={choice.id}>
         <div styleName="caption">
-          <ChoiceLabel choice={c} management={management} paid={isPaidChoice(c)} />
+          <ChoiceLabel choice={choice} management={management} paid={isPaidChoice(choice)} />
         </div>
         <div styleName="labels">
-          {!!c.price && (
+          {!!choice.price && (
             <Label>
-              {c.price} {currency}
+              {choice.price} {currency}
             </Label>
           )}
-          {c.placesLimit === 0 ? null : (
+          {choice.placesLimit === 0 ? null : (
             <PlacesLeft
-              placesLimit={c.placesLimit}
-              placesUsed={placesUsed[c.id] || 0}
-              isEnabled={c.isEnabled}
+              placesLimit={choice.placesLimit}
+              placesUsed={placesUsed[choice.id] || 0}
+              isEnabled={choice.isEnabled}
             />
           )}
         </div>
       </div>
     ),
+    disabled:
+      !choice.isEnabled ||
+      isPaidChoiceLocked(choice) ||
+      (choice.placesLimit > 0 &&
+        (placesUsed[choice.id] || 0) >= choice.placesLimit &&
+        !existingValue[choice.id]),
   }));
 
+  const handleChange = evt => {
+    if (!evt.target.value) {
+      onChange({'': -1});
+      return;
+    }
+    const selected = choices.find(c => c.caption === evt.target.value);
```

Comment:
I don't like this lookup based on the caption... it's basically a roundtrip from unique id to (not guaranteed-to-be-unique) caption to unique id.

Can't we use the option value as the value of the ind-combobox? This would also match the standard HTML `<select>` `<option>` behavior more closely.

And this is functionality we'll certainly need to use this combobox elsewhere since there we typically don't want to have lots of boilerplate code around it to implement option values...

Reply:
> I don't like this lookup based on the caption... it's basically a roundtrip from unique id to (not guaranteed-to-be-unique) caption to unique id.

That lookup has to happen somewhere anyway because this is *not* a select box. Then we again have this interpretation vs physical reality situation we had with "invalid" property.

The difference between combobox and a select box is that a combobox is fully editable, which means user can enter anything they want. If we don't want then, then we want to implement a different widget: searchable select list. That would mean a non-editable label coupled with a text field whose sole purpose is to filter the list.

As for uniqueness of the captions, if you ask me, that absolutely must be guaranteed in the management area, as having two identical captions with different values simply adds confusion for the user. How do you pick one if they are the same??

Reply:
> That lookup has to happen somewhere anyway because this is not a select box

Does it? In the end you create options from a list of options that typically have a label and a value. Couldn't the value of the selected option be stored in a member variable of the ind-combo-box instance or a hidden field? That way you'd never need to derive the value from the caption.

> As for uniqueness of the captions, if you ask me, that absolutely must be guaranteed in the management area, as having two identical captions with different values simply adds confusion for the user. How do you pick one if they are the same??
I agree that we typically do not want non-unique captions. But there are some cases - albeit very rare - where it could happen: Someone selects an option, then that option is removed by a manager (existing registrations keep it!), and later another option with the same caption is added back.

The dropdown will then show the old option (IIRC with an indicator next to the field that you may not be able to select it again if you deselect it and save), but also any new options since those would be valid choices now as well.

> The difference between combobox and a select box is that a combobox is fully editable, which means user can enter anything they want.

Maybe there was a misunderstanding, but that functionality never existed in Indico. At least not in the registration form (we use the SUI Dropdown with `allowAdditions` in some places, typically for an editable list of things like file extensions). The dropdown for the Single Choice registration form field was always supposed to be a (searchable) select list!

Reply:
> Does it? In the end you create options from a list of options that typically have a label and a value.

The intended use case for combobox is basically exactly the same as using a `<datalist>`. It simply provides candidates for the text input. This is why it's allowed for the input to also perform the search. From the perspective of the end user, they are entering a value into the input. 

Select list is not editable because it does not allow arbitrary input. Since it is not editable, there is no expectation that an arbitrary value is allowed.

> Couldn't the value of the selected option be stored in a member variable of the ind-combo-box instance or a hidden field? That way you'd never need to derive the value from the caption.

That's technically doable (and that's similar to what SUI does), but it breaks the user's intuition about how the field works, and this is especially true for people who cannot see the field once they've left it. In order to correct the perception of what's to be expected of the field, the field itself cannot be editable. Therefore, a separate control is provided to facilitate searching and it's part of the popup listbox, not the form control.

> The dropdown will then show the old option (IIRC with an indicator next to the field that you may not be able to select it again if you deselect it and save), but also any new options since those would be valid choices now as well.

In that case, you can't use a combobox. It will be semantically very confused. I recommend using a select list with separate search in the listbox.

Reply:
Then I think there was indeed some misunderstanding somewhere that made you go for a combobox :/

Anyway, I think we can still keep it around in case it becomes useful at some point elsewhere in Indico?

Reply:
Yeah, if you need an actual text input that supports completion candidates, then combobox will work as expected. I'll add a searchable select list to this PR and rewire the SingleSelectInput to use that instead of the combobox and keep the combobox custom element around.

Reply:
Before I do the PR, I'll implement a prototype in a sandbox, tho, and then we can take it from there.

Reply:
@ThiefMaster Check this out:

https://codepen.io/hayavuk/pen/mdYLYEL?editors=0110

It's still not 100% done yet, but it shows the behavior.

In general, it behaves more or less like the native select, with the differences being:

- It automatically expands the list when you press the arrow keys. I think that behavior is ok, but it's actually a hack to work around the issue of changes in value not being announced to screen readers when the list is closed.
- It has a search box.
- It can display arbitrary elements other than just the value label.

Reply:
Works nice! I just got a question regarding the look of the element - are you planning to leave the search input inside the listbox? I'd like to keep the new select list similar to the combobox for uniformity and have the search input on top. What do you think?

Reply:
That doesn't make sense to me. How would you differentiate between selected value and random input?

Reply:
We could just remove it when the field loses focus, like SUI does: https://react.semantic-ui.com/modules/dropdown/#types-search-selection

Reply:
That's the whole problem. You're not supposed to edit the field *after* it loses focus unless it's clear that the field is temporary (e.g., a filter for the list). This is where the whole thing about the original combobox started. SUI's combobox behavior is bad, I tried to fix it, and then it turned out it's bad because it's trying to present a searchable select list as a combobox.

Reply:
Also, I've updated the example to include a way to clear the field. You can add the `required` attribute to the `<x-select>` just like you would to any other control to see its behavior when the field is required.

Reply:
@tomasr8 I re-read your original comment, and I think this is basically what you had in mind:

https://codepen.io/hayavuk/pen/jOoomaN?editors=0100

This actually doesn't affect the separation between selected value caption and the search input. It simply visually places the input "on top" of the control caption so that it appears as if the caption is editable. I used the placeholder instead of populating the value as a work-around for the issues I mentioned above.

Reply:
Just for reference, here's another example from MUI: https://mui.com/material-ui/react-autocomplete/#combo-box

Reply:
Not sure what you mean by "add an extra value". Can you elaborate?